lib/config.service.ts

```diff
@@ -122,6 +122,73 @@ export class ConfigService<
     return defaultValue as T;
   }
 
+  /**
+   * Get a configuration value (either custom configuration or process environment variable)
+   * based on property path (you can use dot notation to traverse nested object, e.g. "database.host").
+   * @param propertyPath
+   */
+  getOrThrow<T = any>(propertyPath: KeyOf<K>): Exclude<T, undefined>;
+  /**
+   * Get a configuration value (either custom configuration or process environment variable)
+   * based on property path (you can use dot notation to traverse nested object, e.g. "database.host").
+   * @param propertyPath
+   * @param options
+   */
+  getOrThrow<T = K, P extends Path<T> = any, R = PathValue<T, P>>(
+    propertyPath: P,
+    options: ConfigGetOptions,
+  ): Exclude<T, undefined>;
+  /**
+   * Get a configuration value (either custom configuration or process environment variable)
+   * based on property path (you can use dot notation to traverse nested object, e.g. "database.host").
+   * It returns a default value if the key does not exist.
+   * If no default value was provided an exception will be thrown.
+   * @param propertyPath
+   * @param defaultValue
+   */
+  getOrThrow<T = any>(
+    propertyPath: KeyOf<K>,
+    defaultValue: NoInferType<T>,
+  ): Exclude<T, undefined>;
+  /**
+   * Get a configuration value (either custom configuration or process environment variable)
+   * based on property path (you can use dot notation to traverse nested object, e.g. "database.host").
+   * It returns a default value if the key does not exist.
+   * If no default value was provided an exception will be thrown.
+   * @param propertyPath
+   * @param defaultValue
+   * @param options
+   */
+  getOrThrow<T = K, P extends Path<T> = any, R = PathValue<T, P>>(
+    propertyPath: P,
+    defaultValue: NoInferType<R>,
+    options: ConfigGetOptions,
+  ): Exclude<R, undefined>;
+  /**
+   * Get a configuration value (either custom configuration or process environment variable)
+   * based on property path (you can use dot notation to traverse nested object, e.g. "database.host").
+   * It returns a default value if the key does not exist.
+   * If no default value was provided an exception will be thrown.
```

Comment:
```suggestion
   * If no default value was provided or if it is `undefined` an exception will be thrown.
```

I guess? just want make clear that behavior

Or maybe would be better if we allow the default value to be `undefined`? :thinking: